kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ResourcesUtil.java

```diff
@@ -277,9 +277,7 @@ static <O extends HasMetadata, R extends HasMetadata> Set<ResourceID> findReferr
                 primary -> refAccessor.apply(primary).stream().anyMatch(ref -> ResourcesUtil.isReferent(ref, referent)));
     }
 
-    static List<Condition> maybeAddOrUpdateCondition(List<Condition> conditions, Condition condition) {
-        var type = Objects.requireNonNull(condition.getType());
-
+    static List<Condition> maybeAddOrUpdateCondition(List<Condition> conditions, List<Condition> conditionsToAdd) {
```

Comment:
worth a javadoc on this one around the `null == remove` logic and how we want to pick the freshest condition